flocker/node/test/test_loop.py

```diff
@@ -557,6 +611,38 @@ def test_convergence_done_delays_new_iteration(self, logger):
              [(NodeStateCommand, dict(state_changes=(local_state,)))])
         )
 
+    def test_convergence_done_delays_new_iteration_ack(self):
+        """
+        An FSM completing the changes from one convergence iteration doesn't
+        start another iteration, if the control node hasn't acknowledged the
+        last state update.
+        """
+        self.local_state = local_state = NodeState(hostname=u'192.0.2.123')
+        self.configuration = configuration = Deployment()
+        self.cluster_state = received_state = DeploymentState(nodes=[])
+        self.action = action = ControllableAction(result=succeed(None))
+        deployer = ControllableDeployer(
+            local_state.hostname, [succeed(local_state)], [action]
+        )
+        client = self.make_amp_client([local_state])
+        reactor = Clock()
+        loop = build_convergence_loop_fsm(reactor, deployer)
+        loop.receive(_ClientStatusUpdate(
+            client=DelayedAMPClient(client), configuration=configuration,
```

Comment:
:question: I don't understand the use of `DelayedAMPClient` here. We never seem to call its `respond` method. Perhaps that's the point, but the fact that it has a `respond` method that we don't use in this branch is a bit confusing. Add a few explanatory comments.